### PR TITLE
Error handling: unknown encoding #145

### PR DESCRIPTION
**Pull request by robindaumann**
Added additional except. Such that a proper error can be reported. Calling sys.exit() because we cannot do anything if the file could not be opened.

### CODE REVIEW

**Comment by jeffkaufman** [comment]
Thanks! Could you add a test?

**Comment by robindaumann** [comment]
Sure. I will add it to the PR.

**Comment by robindaumann** [comment]
Perhaps we should also remove the nightly python build from the checks.

**Comment by robindaumann** [comment]
I wrote the test for the errors that are handled.

File: test.sh [hunk 1]
```diff
@@ -79,6 +79,28 @@ function check_gold() {
   fi
 }
 
+function check_error() {
```
Comment (jeffkaufman) [inline test.sh]: What's the reason for introducing `check_error` instead of using `check_gold`?

Comment (robindaumann) [reply test.sh]: I did not see that there were already tests for the error reporting. I reverted the commit and fixed it. I apologize for that.

**Comment by jeffkaufman** [comment]
> Perhaps we should also remove the nightly python build from the checks.

Travis's "nightly" is stuck on an old version of python which raises a spurious `StopIteration` related error.  It looks like to fix this you need to tell Travis to use Xenial, which is fine: ebd46c5

**Pull request merged**